### PR TITLE
News revamp custom tab

### PR DESCRIPTION
**Pull request by andrewsi10**
Removed all Webviews from the news section, added Chrome Custom Tabs, and revamped GUI. News websites are now preloaded in background (at least they should be).

### CODE REVIEW

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/MainActivity.java [hunk 1]
```diff
@@ -116,7 +117,8 @@ public void onBackPressed() {
             mDrawerLayout.closeDrawer(mDrawerList);
             return;
         }
-
+//        super.onBackPressed();
+        // TODO slight bit of flickering when opening in browser (aka custom chrome tab)
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/MainActivity.java]: Never leave commented out code in the final release, if it's not needed just delete it

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java [hunk 2]
```diff
@@ -1,68 +1,176 @@
...
+        public CustomListAdapter(Context context, String[] newsNames, NewsSite[] news) {
+            super(context, R.layout.fragment_news, newsNames);
+            this.news = news;
+            this.context=context;
+        }
+
+        public View getView(int position,View view,ViewGroup parent) {
+            LayoutInflater inflater = LayoutInflater.from(context);
+            View rowView=inflater.inflate(R.layout.news_list_item, null,true);
+//            View innerView = inflater.inflate(rowRi)
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java]: Same here, unnecessary comments = delete

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java [hunk 3]
```diff
@@ -1,68 +1,176 @@
...
+        builder = new CustomTabsIntent.Builder();
+        share = new Intent(Intent.ACTION_SEND);
+        share.setType("text/plain");
+        builder.setToolbarColor(0x3E50B4);
+        builder.setStartAnimations(getContext(),
+                android.support.design.R.anim.abc_popup_enter,
+                android.support.design.R.anim.abc_popup_exit);
+//                    builder.setExitAnimations(getContext(),
+//                            android.support.design.R.anim.abc_popup_exit,
+//                            android.support.design.R.anim.abc_popup_enter);
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java]: Same

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java [hunk 4]
```diff
@@ -73,40 +181,48 @@ public void onCreate(Bundle savedInstanceState) {
...
-    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
-        inflater.inflate(R.menu.news, menu);
-        super.onCreateOptionsMenu(menu, inflater);
+    private void addNews() {
+        String dpDescription = "The Daily Pennsylvanian is the independent student newspaper of " +
+                "the University of Pennsylvania.";
+        String thirtyFourDescription = "34th Street Magazine is the DP's arts and " +
+                "entertainment weekly magazine.";
+        String utbDescription = "Under The Button is Penn's 24/7 news and entertainment blog, " +
+                "known for its signature humor, gossip and snarky features.";
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java]: Just have them on 1 line, nobody's going to bug you about 80 chars here

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsTab.java [hunk 5]
```diff
@@ -52,7 +50,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     }
 
     public void loadNews() {
-        mIsWebViewAvailable = true;
+//        mIsWebViewAvailable = true;
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsTab.java]: Comments...

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsTab.java [hunk 6]
```diff
@@ -101,15 +89,24 @@ public void onResume() {
...
-    @Override
-    public void onDestroyView() {
-        mIsWebViewAvailable = false;
-        super.onDestroyView();
-    }
+//    @Override
+//    public void onDestroyView() {
+//        mIsWebViewAvailable = false;
+//        super.onDestroyView();
+//    }
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsTab.java]: delete comments

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsTab.java [hunk 7]
```diff
@@ -126,7 +123,7 @@ public void onDestroy() {
     /**
      * Gets the WebView.
      */
-    public WebView getWebView() {
-        return mIsWebViewAvailable ? mWebView : null;
-    }
+//    public WebView getWebView() {
+//        return mIsWebViewAvailable ? mWebView : null;
+//    }
```
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsTab.java]: Delete comments

**Comment by adelq** [comment]
Besides for the minor code quality issues, it works perfectly in practice. Nice work. LGTM after fixes are made.

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java (hunk 4 above)
Comment (adelq) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/NewsFragment.java]: Not needed for LGTM, if you like them that way you do you

**Comment by andrewsi10** [comment]
Cleaned up the code, should be all good now!

**Comment by jasontsh** [comment]
LGTM

**Pull request merged**